### PR TITLE
Added hardening compiler flags to conan profiles

### PR DESCRIPTION
**Pull request by beckerhe**
@dimitry- Can you have a look if these flags are fine?

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,13 +9,15 @@ project(Orbit C CXX)
 if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
   set(STRICT_COMPILE_FLAGS -Werror=all
                            -Werror=float-conversion
+                           -Werror=format-security
```
Comment (dimitry-) [inline CMakeLists.txt:12]: -Wformat is part of -Wall?

Comment (beckerhe) [reply CMakeLists.txt:12]: Yes, but -Wformat is equivalent to -Wformat=1. And -Wformat=1 does NOT include -Wformat-security. We need to specify -Wformat=2 to get -Wformat-security. And we probably want to do that anyway. Should we do it now?

Comment (dimitry-) [reply CMakeLists.txt:12]: yes if it does not involve too much effort.

Comment (beckerhe) [reply CMakeLists.txt:12]: Alright. I'm on it. I don't expect too much of a hassle. If I remember correctly you pretty much removed all c-style printfs.

Comment (beckerhe) [reply CMakeLists.txt]: Ok, it's done. I had to remove the AddLog functions in OrbitGl which haven't been used anyway. Or let's say the formatting feature was not needed. PTAL!

**Pull request merged**